xdsl/builder.py

```diff
@@ -114,6 +121,9 @@ def region(
         else:
             return Builder._region_args(input)
 
+    def implicit(self) -> _ImplicitBuilder:
```

Comment:
Can you add documentation here?
Also, if we are exposing the `_ImplicitBuilder` class, should we make it public then?